src/modules/project/pages1/projectFunding/views/fundingDetails/sections/FundingDetailsUserEmail.tsx

```diff
@@ -1,49 +1,195 @@
-import { Input } from '@chakra-ui/react'
+import { FormControl, FormErrorMessage, Input, InputGroup, InputRightElement, Switch } from '@chakra-ui/react'
+import { debounce } from 'lodash'
+import { useCallback, useEffect, useState } from 'react'
 import { useTranslation } from 'react-i18next'
+import { BsFillCheckCircleFill, BsFillXCircleFill } from 'react-icons/bs'
 
+import Loader from '@/components/ui/Loader'
+import { useAuthContext } from '@/context'
+import { HorizontalFormField } from '@/modules/profile/pages/profileSettings/common/HorizontalFormField'
+import {
+  UserConfigName,
+  UserNotificationType,
+  useUserNotificationSettings,
+} from '@/modules/profile/pages/profileSettings/hooks/useUserNotificationSettings'
 import { useFundingFormAtom } from '@/modules/project/funding/hooks/useFundingFormAtom'
+import { useProjectAtom } from '@/modules/project/hooks/useProjectAtom'
+import { useFollowedProjectsValue } from '@/pages/auth/state'
+import { CardLayout } from '@/shared/components/layouts'
+import { H1 } from '@/shared/components/typography'
 import { Feedback, FeedBackVariant } from '@/shared/molecules'
+import { lightModeColors } from '@/shared/styles'
+import { useUserEmailIsAvailableLazyQuery } from '@/types'
+import { validEmail } from '@/utils'
 
 import { FieldContainer } from '../../../../../../../shared/components/form/FieldContainer'
 
-export const FundingDetailsUserEmail = () => {
+const EMAIL_VALIDATION_STATE = {
+  IDLE: 'IDLE',
+  LOADING: 'LOADING',
+  FAILED: 'FAILED',
+  SUCCEEDED: 'SUCCEEDED',
+}
+
+export const FundingDetailsUserEmailAndUpdates = () => {
   const { t } = useTranslation()
+  const { user } = useAuthContext()
+  const { project } = useProjectAtom()
+  const followedProjects = useFollowedProjectsValue()
+
+  const [followsProject] = useState(followedProjects.find((p) => p.id === project.id) !== undefined)
+  const [userId] = useState(user?.id || 0)
+
+  const { getUserNotificationConfigValue } = useUserNotificationSettings(userId)
+  const subscribedToGeyserEmails =
+    getUserNotificationConfigValue(UserNotificationType.PRODUCT_UPDATES, UserConfigName.IS_ENABLED) === 'true'
+
+  const [emailValidationState, setEmailValidationState] = useState(EMAIL_VALIDATION_STATE.IDLE)
+
+  const showEmailComponent = !user?.email || !followsProject || !subscribedToGeyserEmails
 
   const {
-    formState: { email, needsShipping },
+    formState: { needsShipping, followProject, subscribeToGeyserEmails, email },
     hasSelectedRewards,
     setTarget,
     fundingFormError,
     setErrorstate,
   } = useFundingFormAtom()
 
-  if (!hasSelectedRewards) {
-    return null
+  /*
+   Set the email from the user to the funding form. We do this because the input field
+   is not shown if the user already has an email.
+  */
+  useEffect(() => {
+    if (user?.email) setTarget({ target: { name: 'email', value: user.email } })
+    else setTarget({ target: { name: 'email', value: '' } })
+  }, [user])
```

Comment:
there must be another export from the "useFundingFormAtom" hook, that is "setValue", which could be used for cases like these. setTarget was intended for use inside input field events, where values are found in event.target.value by default.

Reply:
I didn't see one but feel free to add it before merging the PR!